### PR TITLE
introduce Preview screenshot testing with Showkase and Roborazzi

### PR DESCRIPTION
**Pull request by morux2**

### CODE REVIEW

File: app/src/test/java/com/example/screenshotpractice/ShowkaseScreenshotTest.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+@GraphicsMode(GraphicsMode.Mode.NATIVE)
+@Config(qualifiers = RobolectricDeviceQualifiers.MediumTablet)
+class ShowkaseScreenshotTest(
+    private val showkaseBrowserComponent: ShowkaseBrowserComponent,
+) {
+
+    @Test
+    fun previewScreenshot() {
+        val filePath =
+            DEFAULT_ROBORAZZI_OUTPUT_DIR_PATH + "/" + showkaseBrowserComponent.group + "_" + showkaseBrowserComponent.componentName + ".png"
```
Comment (morux2) [inline app/src/test/java/com/example/screenshotpractice/ShowkaseScreenshotTest.kt:25]: スクリーンショット 2023-08-28 20 39 41

**Comment by takahirom** [comment]
@morux2 
Looks great! May I incorporate this information into Roborazzi's documentation or an issue to clarify how to use Roborazzi with Showkase?

**Comment by morux2** [comment]
@takahirom 
Thank you for your message .I have referred to your PR of DroidKaigi.
Of course! I'd be happy to assist with the documentation work.

---
ご連絡ありがとうございます！DroidKaigiのPRを参考に実装させていただきました！
もちろん参照いただいて大丈夫です。もしよろしければドキュメンテーションのお手伝いができたら嬉しいです。

File: app/build.gradle [hunk 2]
```diff
@@ -2,6 +2,7 @@ plugins {
     id 'com.android.application'
     id 'org.jetbrains.kotlin.android'
     id 'io.github.takahirom.roborazzi'
+    id 'com.google.devtools.ksp'
```
Comment (morux2) [inline app/build.gradle:5]: https://github.com/airbnb/Showkase#installation

Comment (morux2) [inline app/build.gradle:5]: > Add the dependency to your module's build.gradle file. If you have a multi-module setup, add this dependency to all the modules with UI elements that should be displayed inside the Showkase browser.
Showkase supports both ksp and kapt. By default, it uses kapt as we only recently added ksp support.

https://github.com/airbnb/Showkase#installation

File: app/src/main/java/com/example/screenshotpractice/ShowkaseScreenshotModule.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.example.screenshotpractice
+
+import com.airbnb.android.showkase.annotation.ShowkaseRoot
+import com.airbnb.android.showkase.annotation.ShowkaseRootModule
+
+@ShowkaseRoot
+class ShowkaseScreenshotModule : ShowkaseRootModule
```
Comment (morux2) [inline app/src/main/java/com/example/screenshotpractice/ShowkaseScreenshotModule.kt:7]: > Define an implementation of the ShowkaseRootModule interface in your root module.

https://github.com/airbnb/Showkase#installation

Comment (morux2) [reply app/src/main/java/com/example/screenshotpractice/ShowkaseScreenshotModule.kt:7]: This is required to call `Showkase.getMetadata()`

File: app/src/test/java/com/example/screenshotpractice/ShowkaseScreenshotTest.kt (hunk 1 above)
Comment (morux2) [inline app/src/test/java/com/example/screenshotpractice/ShowkaseScreenshotTest.kt:25]: スクリーンショット 2023-09-25 12 25 19

you can retrieve the preview name or group name and apply it as the filename.

File: app/src/test/java/com/example/screenshotpractice/ShowkaseScreenshotTest.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        captureRoboImage(
+            filePath,
+        ) {
+            showkaseBrowserComponent.component()
+        }
+    }
+
+    companion object {
+
+        @ParameterizedRobolectricTestRunner.Parameters
```
Comment (morux2) [inline app/src/test/java/com/example/screenshotpractice/ShowkaseScreenshotTest.kt:35]: this is the syntax for parameterized tests in Robolectric.

**Pull request merged**